cmd/kubelet/app/server.go

```diff
@@ -986,18 +994,19 @@ func setContentTypeForClient(cfg *restclient.Config, contentType string) {
 //   2 Kubelet binary
 //   3 Standalone 'kubernetes' binary
 // Eventually, #2 will be replaced with instances of #3
-func RunKubelet(kubeServer *options.KubeletServer, kubeDeps *kubelet.Dependencies, runOnce bool) error {
-	hostname, err := nodeutil.GetHostname(kubeServer.HostnameOverride)
```

Comment:
the two function calls with much smaller input parameters is easier to follow... passing kubelet.Dependencies into a utility method exposes *lots* of potential inputs we don't want to affect node name determination

Reply:
But the actual calculation of nodeName requires a cloud module that depends on *kubelet.Dependencies. The previous calculation code is as follows


```
hostName, err := nodeutil.GetHostname(hostnameOverride)
if err != nil {
    return nil, err
}
nodeName, err := getNodeName(kubeDeps.Cloud, hostName)
if err != nil {
    return nil, err
}
```

I think there is a small place that can be optimized. The call link of the whole function only needs to calculate the nodeName once, which can be passed in by parameters.

As @mattjmcnaughton said ,the calculation process is not expensive, just reducing the duplicate code

Reply:
> But the actual calculation of nodeName requires a cloud module that depends on *kubelet.Dependencies.

That's true, but constraining the parameter to the cloud interface itself still limits the surface area exposed to the method that calculates the name, which seems better. 

Separating that from the method that involves environmental hostname detection allows for easier unit testing as well.

Reply:
Thanks , i got it. 

I close the pr ?